gui/src/pages/NewProjectConfig.svelte

```diff
@@ -0,0 +1,150 @@
+<script lang="ts">
+  import Layout from '../components/Layout.svelte';
+  import Textbox from '../components/Textbox.svelte';
+  import Spinner from '../components/Spinner.svelte';
+  import ErrorLabel from '../components/ErrorLabel.svelte';
+  import SubmitButton from '../components/form/SubmitButton.svelte';
+  import CenterFormPanel from '../components/form/CenterFormPanel.svelte';
+  import { api, isClientError } from '../lib/api';
+  import * as router from 'svelte-spa-router';
+
+  export let params = { starter: '' };
+
+  const { starter } = params;
+
+  let name = starter;
+  let templateUrl: string | null = null;
+  let error: Error | null = null;
+
+  let workingDirectory: string | null = null;
+
+  const setWorkingDirectory = (dir: string) => {
+    workingDirectory = dir;
+  };
+
+  (async () => {
+    setWorkingDirectory(await api.kernel.getUserHomeDir());
+
+    if (starter !== 'empty') {
+      templateUrl = String(
+        (await api.kernel.describeTemplates()).find((x) => x.name === starter)
+          ?.url,
+      );
+    }
+  })();
+</script>
+
+<Layout>
+  <CenterFormPanel title={`New ${starter} project`} backRoute="#/new-project">
+    <form
+      on:submit|preventDefault={async () => {
+        error = null;
+        let workspaceId;
+
+        try {
+          workspaceId = await api.kernel.createProject(
+            `${workingDirectory}/${name}`,
+            templateUrl,
+          );
+          router.push(`/workspaces/${encodeURIComponent(workspaceId)}`);
+        } catch (ex) {
+          if (!isClientError(ex)) {
+            throw ex;
+          }
+          error = ex;
+        }
+      }}
+    >
+      <h1>New {starter} project</h1>
+
+      <label for="name">Name:</label>
+      <Textbox
+        bind:value={name}
+        name="name"
+        id="name"
+        placeholder={starter}
+        --input-width="100%"
+      />
+      <div style="height:32px" />
+      {#if workingDirectory}
+        <label for="root">Root:</label>
+        <h2><span>{workingDirectory}<span>/{name}</span></span></h2>
```

Comment:
This breaks when you navigate up to the root directory.

Reply:
I _think_ it's fixed now, thanks @BenElgar